### ISSUE TITLE
[Problem] Geometry selection for FEM features is unintuitive and inconsistent

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#11666]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Geometry selection for FEM features definitely needs some improvements. Let's take [FEM ConstraintForce](https://wiki.freecad.org/FEM_ConstraintForce) as an example:
- GUI says "Select multiple face(s), click Add or Remove" - this is the first problem because the order suggested by this message is incorrect - one has to click the Add button first and then select a geometrical entity
- to make a selection for Direction, one has to click the Add button again to deactivate it, then select a face/edge and press the Direction button
- 
Now let's take a look at [FEM ConstraintContact](https://wiki.freecad.org/FEM_ConstraintContact):
- GUI says "Select master face, click Add or Remove" for the first selection and the suggested order is correct this time - if you click the Add button first, it shows an error message ("Nothing selected!")
- to fill the second selection field, there's no need to deactivate the first Add button, one just has to select another face and click the second Add button (likely because only one face can be selected for each field, unlike in the case of FEM ConstraintForce)

So the first thing to do would be to correct the GUI tips for features like FEM ConstraintForce so that they don't suggest an incorrect selection order. But IMO it's also necessary to unify the selection for all features and make it as user-friendly as possible.

### Full version info

```shell
OS: Windows 11 build 22621
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35274 (Git)
Build type: Release
Branch: main
Hash: dc063eabec61bffc72c92595c66a4649f25fe381
Python 3.10.13, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.6, OCC 7.6.3
Locale: Polish/Poland (pl_PL)
```

### Subproject(s) affected?

FEM

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
It's not even consistent in FEM. Some tools require a selection before you click on the tool, otherwise the user gets an error. Other tools do not work with a pre-selection and you have to select all again with the tool active. In another tool you always have to click the *select* button between each new selection.
It should be:
1. all tools in FEM should use the pre-selection as input where possible.
2. with an open task dialog and option to select elements it should be like the Fillet tool in PartDesign WB. A mode to just select and deselect, of if possible not even a special mode. Just see all selected andeither click on them to deselect or select some elements which are not selected.

**Comment by FEA-eng** [comment]
@maxwxyz Right, FEM suffers from extreme inconsistency in this regard. But geometry selection in this workbench is one of the biggest problems in general - you can't select interior entities or multiple entities at once (with a selection box), often making selection extremely tedious. But we have separate issues for that so let's go back to this topic. IMO an ideal approach would be what is done in many other FEA preprocessors - you just start a tool and select geometry, holding SHIFT to add and CTRL to remove items. Of course, they are highlighted but there's no visible list like in FreeCAD. I know that it's not really the FreeCAD way though so I don't mind a list but it should be way more user-friendly and consistent, as you described.

**Comment by maxwxyz** [comment]
This problem is also in the rigid body constraint task dialog.

**Comment by FEA-eng** [comment]
> This problem is also in the rigid body constraint task dialog.

It uses the same selection panel as the displacement or fixed constraint and they should share the same rules in this regard. But in the future, selection should be unified and simplified for other tools. It's the worst for contact/tie, force and such. Boundary conditions (the aforementioned ones and others) are not that bad.
